learn_django/trip_project/views.py

```diff
@@ -1,18 +1,39 @@
-
-from django.shortcuts import render
+from django.contrib import messages
+from django.shortcuts import render, redirect
 from rest_framework import viewsets
 
+from .forms import UserForm, ProfileForm
 from .models import Trip
 from .serializers import TripSerializer
+from .permissions import IsSuperUserOrStaff
 
 
-class TripView(viewsets.ModelViewSet):
+class TripsView(viewsets.ModelViewSet):
     queryset = Trip.objects.all()
     serializer_class = TripSerializer
+    permission_classes = [IsSuperUserOrStaff]
+
+
+def register(request):
+    if request.method == 'POST':
+        user_form = UserForm(request.POST)
+        profile_form = ProfileForm(request.POST)
+        if user_form.is_valid() and profile_form.is_valid():
+            user = user_form.save()
```

Comment:
profile should be created on the `save`. correct? 

why `profile_form.user` is required?